tests/integration/defs/disaggregated/test_disaggregated_etcd.py

```diff
@@ -0,0 +1,426 @@
+                            stderr=subprocess.PIPE,
+                            text=True)
+
+    if result.returncode == 0:
+        logger.info("Client test succeeded")
+        logger.info(f"Client output: {result.stdout}")
+        return True
+    else:
+        logger.error(f"Client test failed with return code {result.returncode}")
+        logger.error(f"Error output: {result.stderr}")
+        logger.error(f"Standard output: {result.stdout}")
+        return False
+
+
+def kill_server_by_port(port: int) -> bool:
+    """Find and kill a process by port using lsof."""
+    try:
+        # Find PID using port
+        cmd = ["lsof", "-t", f"-i:{port}"]
+        result = subprocess.run(cmd, stdout=subprocess.PIPE, text=True)
+
+        if result.stdout.strip():
+            pid = int(result.stdout.strip())
+            os.kill(pid, signal.SIGKILL)
+            logger.info(f"Killed process {pid} on port {port}")
+            return True
+        else:
+            logger.warning(f"No process found on port {port}")
+            return False
+    except Exception as e:
+        logger.error(f"Error killing process on port {port}: {e}")
+        return False
+
+
+def cleanup_processes(processes):
+    """Kill all started processes."""
+    logger.info("Cleaning up all processes...")
+
+    for name, process in processes.items():
+        if process.poll() is None:  # Still running
+            logger.info(f"Terminating {name} (PID: {process.pid})")
+            try:
+                process.terminate()
+                process.wait(timeout=5)
+            except (subprocess.TimeoutExpired, ProcessLookupError):
+                logger.warning(f"Force killing {name} (PID: {process.pid})")
+                try:
+                    process.kill()
+                except ProcessLookupError:
+                    pass
+
+
+def start_etcd_server(working_dir, env=None) -> subprocess.Popen:
+    """Start etcd server."""
+    cmd = ["etcd"]
+
+    logger.info("Starting etcd server...")
+    process = subprocess.Popen(cmd,
+                               env=env,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               text=True,
+                               bufsize=1,
+                               cwd=working_dir)
+    return process
+
+
+def cleanup_etcd_data(env=None):
+    """Clean up etcd data using etcdctl."""
+    cmd = ["etcdctl", "del", "--prefix", "trtllm/"]
+
+    logger.info("Cleaning etcd data...")
+    result = subprocess.run(cmd,
+                            env=env,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            text=True)
+
+    if result.returncode == 0:
+        logger.info("Successfully cleaned etcd data")
+    else:
+        logger.warning(f"Failed to clean etcd data: {result.stderr}")
+
+
+def create_config_files(config):
```

Comment:
Would it be possible to use an existing file here instead of writing one dynamically, to improve maintainability?